services/bettersocial/user/BetterSocialBlockUserV2.js

```diff
@@ -13,6 +13,91 @@ const Getstream = require('../../../vendor/getstream');
 const RedisBlockHelper = require('../../redis/helper/RedisBlockHelper');
 const BetterSocialScoreBlockUser = require('../score/block-user');
 
+const getstream_unfollow_sequence = async (token, feedOwner, feedTarget, direct = false) => {
+  try {
+    // Main feed following unfollow user anon feed target
+    if (direct) {
+      await Getstream.feed.unfollowMainFeedFollowingDirect(feedOwner, feedTarget);
+    } else {
+      await Getstream.feed.unfollowMainFeedFollowing(token, feedOwner, feedTarget);
+    }
+    // Main feed unfollow user feed target
+    await Getstream.feed.unfollowUser(feedOwner, feedTarget);
+    // Main feed unfollow user exclusive feed target
+    await Getstream.feed.unfollowUserExclusive(feedOwner, feedTarget);
+  } catch (e) {
+    console.log('Error in block user v2 getstream');
+    console.log(e);
+    return {
+      isSuccess: false,
+      message: e?.message || 'Error in block user v2 getstream'
+    };
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `getstream_unfollow_sequence` does not have a return statement in the successful execution path. This could lead to unexpected behavior as the caller might expect a response object similar to the one returned in case of an error. Consider adding a success response.

```diff
  try {
    // Main feed following unfollow user anon feed target
    if (direct) {
      await Getstream.feed.unfollowMainFeedFollowingDirect(feedOwner, feedTarget);
    } else {
      await Getstream.feed.unfollowMainFeedFollowing(token, feedOwner, feedTarget);
    }
    // Main feed unfollow user feed target
    await Getstream.feed.unfollowUser(feedOwner, feedTarget);
    // Main feed unfollow user exclusive feed target
    await Getstream.feed.unfollowUserExclusive(feedOwner, feedTarget);
  } catch (e) {
    console.log('Error in block user v2 getstream');
    console.log(e);
    return {
      isSuccess: false,
      message: e?.message || 'Error in block user v2 getstream'
    };
  }
+ return {
+   isSuccess: true,
+   message: 'Unfollow sequence completed successfully'
+ };
};
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->